scripts/utils.sh

```diff
@@ -262,4 +248,51 @@ validate_native_address() {
   echo "${native_address}"
 }
 
+# $1 = endpoint
+get_authenticated_curl_cmd() {
+  # the following checks must handle endpoints like:
+  #   - myendpoint.com
+  #   - myendpoint.com:3001
+  #   - apitoken@myendpoint.com
+  #   - apitoken@myendpoint.com:3001
+  #   - http(s)://myendpoint.com
+  #   - http(s)://myendpoint.com:3001
+  #   - http(s)://apitoken@myendpoint.com
+  #   - http(s)://apitoken@myendpoint.com:3001
+  local full_endpoint="${1}"
+
+  # extract protocol prefix incl. separator ://
+  local protocol="$(echo ${full_endpoint} |
+    grep :// | sed -e's,^\(.*://\).*,\1,g')"
```

Comment:
Beware that the regex also accepts leading whitespace and empty protocol (`://`) into the capture group. Should be something like `([^\r\n\s]+://)`.

Reply:
Adapted, now whitespace is trimmed. The empty protocol is expected and used later to remove the prefix.